src/main/scala/com/gu/SlackIncomingWebHook.scala

```diff
@@ -1,34 +1,24 @@
 package com.gu
 
-import org.apache.http.client.methods.{CloseableHttpResponse, HttpPost}
-import org.apache.http.entity.StringEntity
-import org.apache.http.impl.client.HttpClients
-import play.api.libs.json.{JsValue, Json}
-import scala.io.Source
-
-class SlackIncomingWebHook(url: String) extends Http {
-
-  def send(payload: Payload): HttpResponse = {
-    POST(url, Json.toJson(payload))
-  }
-
-}
-
-trait Http {
-
-  val client = HttpClients.createDefault()
-
-  def POST(url: String, payload: JsValue): HttpResponse = {
-    val post = new HttpPost(url)
-
-    post.setHeader("Content-type", "application/x-www-form-urlencoded")
-    post.setEntity(new StringEntity(s"payload=${payload.toString}"))
-
-    new HttpResponse(client.execute(post))
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import play.api.libs.json.Json
+import play.api.libs.ws.WSResponse
+import play.api.libs.ws.ahc.AhcWSClient
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class SlackIncomingWebHook(url: String) {
```

Comment:
This looks really appealing for my current use case for this library, as will be using it in a Play app, so can just use the dependency injection for WS etc. Makes it less self contained for other projects though